Service/app/models/aci/definitions.py

```diff
@@ -27,15 +27,22 @@ def before_definition_update(filters, data, **kwargs):
         check_all_mo_exist(data["managed_objects"])
     return (filters, data)
 
+def before_definition_delete(filters, **kwargs):
+    obj = Definitions.load(_filters = filters , **kwargs)
+    if obj.template == True:
+        abort( 400, "Cannot delete a predefined template")
+    return filters
+
```

Comment:
this isn't going to work after we merge rest changes.  You need to load the specific obj if present.  For example:
```
if "definition" in filters:
   obj = Defintion.load(defintion=filters["defintion"])
   if obj.exists() and obj.template is True:
        ...abort
```